### PR TITLE
fix(run): add ctx to run.Store

### PR DESCRIPTION
**Pull request by Arqu**
Also added an `owner profile.ID` to `ListByStatus` as cloud will have to scope it down and from a multi tenancy standpoint makes sense to control for it. And I want to avoid hacks we used to do where we magically pull out the relevant user from context.

### CODE REVIEW

File: automation/run/store.go [hunk 1]
```diff
@@ -25,26 +27,26 @@ var (
...
 	// workflow.ID
-	GetStatus(wid workflow.ID) (Status, error)
+	GetStatus(ctx context.Context, wid workflow.ID) (Status, error)
 	// ListByStatus returns a list of run.State entries with a given status
 	// looking only at the most recent run of each Workflow
-	ListByStatus(s Status, lp params.List) ([]*State, error)
+	ListByStatus(ctx context.Context, owner profile.ID, s Status, lp params.List) ([]*State, error)
 	// Shutdown closes the store
-	Shutdown() error
+	Shutdown(ctx context.Context) error
```
Comment (b5) [inline automation/run/store.go:49]: wait, do we _really_ need a context on shutdown? Strikes me as a bad idea

**Review by b5 — COMMENTED** [review]
I'm not _so_ into adding contexts everywhere, but see how this is an important stop-gap in the short term. If we can nix it from the shutdown method, looks good

File: automation/run/store.go (hunk 1 above)
Comment (Arqu) [reply automation/run/store.go]: Probably not, cloud should have a separate shutdown mechanic anyways (and does).

**Comment by Arqu** [comment]
Updated the PR with a bit more stuff, the `step.Duration` can exceed a 32bit int and was giving me some trouble, so aligned that and also removed `ctx` from the `AddEvent` handler as that needs to run on a separate context from the event caller.

**Review by b5 — APPROVED** [review]

**Pull request merged**